### ISSUE TITLE
List structure: definition lists (WCAG 2.2 Level A - Accessibility)

### ISSUE THREAD

**Issue by daniellecorke** [ONSdigital/design-system#3028]
### Background
The eQ service identified this issue within the ONS Design System following their DAC audit report (18/01/2024). 

### Issue 
List structure not properly represented to screen reader users within the markup.

![Image](https://github.com/ONSdigital/design-system/assets/102512949/574b4fe3-4822-4a41-9b82-b48e530e9b3b)

Definition lists have been used incorrectly on this page to provide the company list, meaning that screen reader users may find it difficult to interpret the relationships and structure being presented. 
Definition lists represent a list of key-value pairs, where `<dt>` elements represent the key (definition term), while `<dd>` elements represent the value (definition description). 
Each company name on this page is contained within its own definition list, containing only a `<dt>` element with no `<dd>`, however each `<dt>` element must have an associated `<dd>` element to form a valid definition list.

### Suggested solution
Ensure that list structure is properly represented to screen reader users so that the information can be understood in the same way as by sighted users. Based on the visual appearance of this content, it does not appear that a definition list is necessary, and a bulleted list may be more appropriate as there are no key-value pairs.

### Tasks
- Investigate whether a change can be made and identify any impacts
- Decide the best solution
- Implement the solution
- Update eQ with progress

### Info
- [eQ DAC reports/docs](https://confluence.ons.gov.uk/display/SDC/eQ+DAC+Reports)
- Issue ID: DAC_List_Structure_02
- DAC report pages: 20 to 21

**Comment by rmccar** [comment]
Look into any issues with the implementation of a hub like summary with no links in eQ

**Comment by rmccar** [comment]
> @rmccar Can you please update us on this ticket? Is this ticket needed or a new one needs to be created? @daniellecorke

This is related to the [this ticket](https://github.com/ONSdigital/design-system/issues/3030) and the solution will look to address both, so the update is the same, we should keep this ticket open and there is no update at the moment

**Comment by rmccar** [comment]
I have done some research into this and spoken to DAC about the proposed solution. Out of that I think the best solution is to make changes to the summary macro so that if there is only one item in each row in a `<dt>` or `<dd>` then this should be marked up as a `<ol>` and not a `<dl>`. If there are rows that have both `<td>`and `<dd>` elements but also some that only one or the other `<dt>` or `<dd>` then on the row that only has one of them we add the one that is missing but visually hide it and allow the user to set some descriptive text within that element. This will need some discussion on the best way of achieving this. Also as this is a different solution to [this ticket](https://github.com/ONSdigital/design-system/issues/3030) then I think we should now treat these tickets separately.

**Comment by rmccar** [comment]
There is a spike on eQ to look into whether there is any instances that have a mixture of rows with one and rows with two items in a summary. The outcome of this spike will have an impact on the solution for this card. If there are no instances the solution will be simpler and we just need to mark summaries up as lists if there is only one item and definition lists if there are two. It is only if there are instances of this that we will need to think about implementing visually hidden items to add items to rows with only one item. So this card is blocked on that spike for now

**Comment by rmccar** [comment]
The spike for the investigation into this in eQ is now complete and there shouldn't be any instances of summaries with a mixture of 1 and 2/or more items in rows. Therefore this should be a simple DS change to update the summary component to be marked up as a definition list if there are 2 or more items on rows or an unordered list if there is only 1 item on each row.

**Comment by alessioventuriniAND** [comment]
Having looked into the summary component and how it has been used in the example above I can confirm that the only way to recreate the above example is by passing an empty object to the rowItems parameter. If the rowItems parameter is removed the component breaks. 

As the above example is only displaying a list of companies it seems like the summary component is not the right fit for this purpose. 
Below is the component that is perfect fro displaying a list of names or text : 

```
{% set listItemClasses = "ons-u-pt-s ons-u-pb-s ons-u-m-no ons-u-bb" %}
{% set lastlistItemClasses = "ons-u-pt-s ons-u-pb-s ons-u-m-no" %}
{{
    onsList({
        "variants": 'bare',
        "itemsList": [
            {
                "text": 'What are the dates of the sales period you are reporting for?',
                "listClasses": listItemClasses
            },
            {
                "text": 'For the period 1 January 2015 to 2 February 2017, what was the value of your total turnover, excluding VAT?',
                "listClasses": listItemClasses
            },
            {
                "text": 'Please indicate the reasons for any changes in the total turnover',
                "listClasses": lastlistItemClasses
            }
        ]
    })
}} 
```
